lib/srv/sess.go

```diff
@@ -1787,6 +1802,9 @@ func (s *session) trackerRemoveParticipant(participantID string) error {
 }
 
 func (s *session) trackerUpdateState(state types.SessionState) error {
+	s.state = state
```

Comment:
We should hold a lock on `s.stateUpdate.L` before mutating state. Can you confirm this is done correctly? (a little hard to tell in the diff view)

Reply:
This seems correct, I added one case where it was missing below after my re-review